explorer/src/components/JupiterSwapModal.tsx

```diff
@@ -0,0 +1,562 @@
+					...prevValue,
+					inputMint: new PublicKey(token.address),
+				};
+			});
+
+			let publicKey = wallet.publicKey || new PublicKey(token.address);
+			const info = await connection.getParsedTokenAccountsByOwner(publicKey, {
+				mint: new PublicKey(token.address),
+			});
+			const balance = info.value[0].account.data.parsed.info.tokenAmount.uiAmount;
+			let inputBalance = 0;
+			if ((balance) - 0.00001 > 0) {
+				inputBalance = (balance - 0.00001);
+			}
+			setExchangeAmount(parseFloat((inputBalance * (10 ** token?.decimals)).toString()) || 1);
+			setInputMint(new PublicKey(token.address));
+			setInputToken(token);
+		} else if (tokenSearchType === 'receive' || type === 'receive') {
+			setFormValue((prevValue) => {
+				return {
+					...prevValue,
+					outputMint: new PublicKey(token.address),
+				};
+			});
+			setWalletBalance(token.address, "receive");
+			setOutputMint(new PublicKey(token.address));
+			setOutputToken(token);
+		}
+		setShowTokenSearch(false);
+	};
+
+	const onTokenSearchChange = (value: string) => {
+		setTokenSearch(value);
+	}
+
+	useEffect(() => {
+		const options = allTokens.filter((token) => {
+			const searchLower = tokenSearch.toLowerCase();
+			return (
+				token.name.toLowerCase().includes(searchLower) ||
+				token.symbol.toLowerCase().includes(searchLower) ||
+				token.address.toLowerCase() === searchLower
+			);
+		})
+		setTokens(options);
+	}, [tokenSearch, allTokens]);
+
+	const onAggregatorSearchChange = (value: string) => {
+		setAggregatorSearch(value);
+	}
+
+	useEffect(() => {
+		const options = aggregators.filter((aggregator) => {
+			const searchLower = aggregatorSearch.toLowerCase();
+			return aggregator.name.toLowerCase().includes(searchLower);
+		})
+		setAggregatorOptions(options);
+	}, [aggregatorSearch]);
+
+	const onClickSlippageOptionBtn = (selectedIdx: number) => {
+		setSelectSlippage(slippageOptions[selectedIdx]);
+	};
+
+	const saveSlippageSettings = () => {
+		if (selectSlippage || inputSlippage.length) {
+			setFormValue((prevValue) => {
+				return {
+					...prevValue,
+					slippage: selectSlippage ?? Number(inputSlippage),
+				};
+			});
+			setShowSlippageSettings(false);
+		}
+	};
+
+	const jupiter = useJupiter({
+		amount: exchangeAmount,
+		inputMint,
+		outputMint,
+		slippage: 1,
+		debounceTime: 250,
+	});
+	const {
+		allTokenMints,
+		routeMap,
+		exchange,
+		refresh,
+		lastRefreshTimestamp,
+		loading,
+		routes,
+		error,
+	} = jupiter;
+	useEffect(() => {
+		if (routes) {
+			setDisplayRoutes(routes?.slice(0, 2));
+			setAdditionalRoutes(routes?.slice(2, routes.length));
+		}
+	}, [routes]);
+	const setTokenSearchValues = (type: any) => {
+		setTokenSearchType(type);
+		setShowTokenSearch(true)
+	}
+	const getTokenSymbol = (tokenAddress: String) => {
+		let symbol = null
+		for (let i = 0; i < allTokens.length; i++) {
+			if (allTokens[i].address === tokenAddress) {
+				symbol = allTokens[i].symbol
+				break;
+			}
+		}
+		return symbol;
+	}
```

Comment:
Wrapping this in a useMemo would be preferred as it ensures that the code does not re run when the same tokenAddress is passed in again and also to ensure that the reference to the allTokens state is updated.

```suggestion
	const getTokenSymbol = useMemo((tokenAddress: String) => {
	        const tokenObj = allTokens.filter(token => token.address === tokenAddress)
	        return tokenObj.length > 0 ? tokenObj[0].symbol : null
	}, [allTokens])
```